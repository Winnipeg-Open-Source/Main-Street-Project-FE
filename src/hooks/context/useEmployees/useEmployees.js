import { useContext } from 'react';
import EmployeesContext from 'context/Employees';

function useEmployees () {
    return useContext(EmployeesContext);
}

export default useEmployees;
