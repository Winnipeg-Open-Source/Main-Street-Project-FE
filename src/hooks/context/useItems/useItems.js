import { useContext } from 'react';
import ItemsContext from 'context/Items';

function useItems () {
    return useContext(ItemsContext);
}

export default useItems;
