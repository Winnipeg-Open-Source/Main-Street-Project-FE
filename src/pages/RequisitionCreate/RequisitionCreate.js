import React, { useEffect } from 'react';
import Page from 'components/Page';
import RequisitionForm from 'containers/RequisitionForm';
import useEmployees from 'hooks/context/useEmployees';
import useItems from 'hooks/context/useItems';

function RequisitionCreate () {
    const { isLoading: isLoadingEmployees, onLoadEmployees } = useEmployees();
    const { isLoading: isLoadingItems, onLoadItems } = useItems();

    useEffect(() => {
        isLoadingEmployees && onLoadEmployees();
        isLoadingItems && onLoadItems();
    }, []);

    return (
        <Page title='New Requisition'>
            <RequisitionForm />
        </Page>
    );
}

RequisitionCreate.displayName = 'RequisitionCreate';

export default RequisitionCreate;
