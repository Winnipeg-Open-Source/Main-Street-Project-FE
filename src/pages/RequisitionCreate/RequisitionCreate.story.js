import React from 'react';
import StorybookRouter from 'storybook/helpers/StorybookRouter';
import MockProviders from 'storybook/helpers/MockProviders';
import mockAxios from 'storybook/helpers/mockAxios';
import RequisitionCreate from './RequisitionCreate';
import { mockEmployees } from 'storybook/mocks/employees';
import { mockItems } from 'storybook/mocks/items';
import {
    EMPLOYEES_API_PATH,
    ITEMS_API_PATH,
} from 'constants/api';

export const Loading = () => {
    mockAxios.onGet(EMPLOYEES_API_PATH).reply(200, mockEmployees);
    mockAxios.onGet(ITEMS_API_PATH).reply(200, mockItems);

    return (
        <StorybookRouter>
            <MockProviders>
                <RequisitionCreate />
            </MockProviders>
        </StorybookRouter>
    );
};

export const Loaded = () => {
    return (
        <StorybookRouter>
            <MockProviders>
                <RequisitionCreate />
            </MockProviders>
        </StorybookRouter>
    );
};

export default {
    title: 'Pages|RequisitionCreate',
    component: RequisitionCreate,
};
