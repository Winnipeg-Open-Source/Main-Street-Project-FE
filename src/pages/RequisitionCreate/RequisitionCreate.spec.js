import React from 'react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import RequisitionCreate from './RequisitionCreate';
import useEmployees from 'hooks/context/useEmployees';
import useItems from 'hooks/context/useItems';

jest.mock('hooks/context/useEmployees');
jest.mock('hooks/context/useItems');

describe ('RequisitionCreate', () => {
    it ('renders without crashing', () => {
        const mockLoadEmployees = jest.fn();
        const mockLoadItems = jest.fn();

        useEmployees.mockImplementation(() => ({ isLoading: true, onLoadEmployees: mockLoadEmployees }));
        useItems.mockImplementation(() => ({ isLoading: true, onLoadItems: mockLoadItems }));

        const { asFragment } = renderWithThemeAndRouter(<RequisitionCreate />);

        expect(mockLoadEmployees).toHaveBeenCalledTimes(1);
        expect(mockLoadItems).toHaveBeenCalledTimes(1);
        expect(asFragment()).toMatchSnapshot();
    });

    it ('does not load already loaded resources', () => {
        const mockLoadEmployees = jest.fn();
        const mockLoadItems = jest.fn();

        useEmployees.mockImplementation(() => ({ isLoading: false, onLoadEmployees: mockLoadEmployees }));
        useItems.mockImplementation(() => ({ isLoading: false, onLoadItems: mockLoadItems }));

        renderWithThemeAndRouter(<RequisitionCreate />);

        expect(mockLoadEmployees).toHaveBeenCalledTimes(0);
        expect(mockLoadItems).toHaveBeenCalledTimes(0);
    });
});
