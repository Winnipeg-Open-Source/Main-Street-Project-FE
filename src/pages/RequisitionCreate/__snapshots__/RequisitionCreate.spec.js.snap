// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequisitionCreate renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-AxhCb sc-pAMRk cPkxDf"
      width="1"
    >
      <div
        class="sc-AxirZ sc-AxhCb iEMUrT"
        width="1"
      >
        <div
          class="sc-AxhUy sc-pIITJ gbtPxP"
          font-size="3"
        >
          New Requisition
        </div>
      </div>
      <div
        class="sc-AxirZ sc-AxhCb sc-pZCbX kEIMlE"
      >
        <div
          class="sc-AxirZ sc-AxhCb sc-oTaAA hapUIm"
        >
          <label
            class="sc-ptfmh cDYtMS className"
            color="border.light"
            font-size="10px"
            font-weight="bold"
          >
            <div
              class="sc-AxirZ sc-AxhCb czMtkd"
              width="1"
            >
              <select
                class="sc-plVHf jbMuUc sc-ptSuy sc-paWCZ bICCST"
                data-testid="employee-select"
                disabled=""
                font-size="2,,1"
                id="Employee / Volunteer"
              >
                <option
                  disabled=""
                  hidden=""
                  value=""
                >
                  Loading...
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="sc-AxmLO fTwYWQ sc-fzqAui sc-pcZmk fxlvpJ"
                color="text.light"
                fill="currentcolor"
                focusable="false"
                height="24"
                role="img"
                tabindex="-1"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
                />
              </svg>
            </div>
          </label>
          <label
            class="sc-ptfmh ibAoZD"
            color="border.light"
            font-size="1"
            font-weight="bold"
          >
            Description
            <textarea
              class="sc-pTUxa sc-qYVps gaUhZc"
              id="requisition-description"
            />
          </label>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            role="combobox"
            style="position: relative; width: 100%;"
          >
            <input
              aria-autocomplete="list"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="sc-pdkfH jYnISK sc-qQlyE jmRcib"
              disabled=""
              font-size="2,,1"
              id="downshift-0-input"
              placeholder="Loading..."
              value=""
              width="1"
            />
            <div
              aria-labelledby="downshift-0-label"
              class="sc-AxirZ sc-pQfvp sc-pjTIO cGtXuM"
              height="256px"
              id="downshift-0-menu"
              role="listbox"
            />
          </div>
          <div
            class="sc-AxirZ sc-AxhCb sc-qYhBd fFHzQn"
            width="1"
          />
        </div>
        <div
          class="sc-AxirZ sc-AxhCb sc-pRFZy eQXfCz"
        >
          <button
            class="sc-ptEpz beoNIQ"
            color="tertiary"
            width="1"
          >
            Cancel
          </button>
          <button
            class="sc-ptEpz bUlphT"
            color="primary"
            width="1"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
